HeterogeneousCore/AlpakaCore/interface/CopyToDeviceCache.h

```diff
@@ -0,0 +1,120 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_CopyToDeviceCache_h
+#define HeterogeneousCore_AlpakaInterface_interface_CopyToDeviceCache_h
+
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaCore/interface/QueueCache.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/devices.h"
+
+namespace cms::alpakatools {
+  namespace detail {
+    // Queue to use for each non-CPU device type
+    template <typename TDevice>
+    struct QueueForCopy;
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    template <>
+    struct QueueForCopy<alpaka::DevCudaRt> {
+      using type = alpaka::QueueCudaRtNonBlocking;
+    };
+#endif
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+    template <>
+    struct QueueForCopy<alpaka::DevHipRt> {
+      using type = alpaka::QueueHipRtNonBlocking;
+    };
+#endif
```

Comment:
Actually, you can use directly below
```
      using Device = TDevice;
      using Queue = alpaka::Queue<Device, alpaka::NonBlocking>;
```
and rely on the fact that we will not call it for the CPU device.

Reply:
Oh cool, thanks.